packages/google_maps_flutter/google_maps_flutter_android/android/src/main/java/io/flutter/plugins/googlemaps/GroundOverlaysController.java

```diff
@@ -0,0 +1,139 @@
+    this.bitmapDescriptorFactoryWrapper = bitmapDescriptorFactoryWrapper;
+  }
+
+  void setGoogleMap(GoogleMap googleMap) {
+    this.googleMap = googleMap;
+  }
+
+  void addGroundOverlays(@NonNull List<Messages.PlatformGroundOverlay> groundOverlaysToAdd) {
+    for (Messages.PlatformGroundOverlay groundOverlayToAdd : groundOverlaysToAdd) {
+      addGroundOverlay(groundOverlayToAdd);
+    }
+  }
+
+  void changeGroundOverlays(@NonNull List<Messages.PlatformGroundOverlay> groundOverlaysToChange) {
+    for (Messages.PlatformGroundOverlay groundOverlayToChange : groundOverlaysToChange) {
+      changeGroundOverlay(groundOverlayToChange);
+    }
+  }
+
+  void removeGroundOverlays(@NonNull List<String> groundOverlayIdsToRemove) {
+    for (@NonNull String groundOverlayId : groundOverlayIdsToRemove) {
+      removeGroundOverlay(groundOverlayId);
+    }
+  }
+
+  @Nullable
+  GroundOverlay getGroundOverlay(@NonNull String groundOverlayId) {
+    GroundOverlayController groundOverlayController =
+        groundOverlayIdToController.get(groundOverlayId);
+    if (groundOverlayController == null) {
+      return null;
+    }
+    return groundOverlayController.getGroundOverlay();
+  }
+
+  private void addGroundOverlay(@NonNull Messages.PlatformGroundOverlay platformGroundOverlay) {
+    GroundOverlayBuilder groundOverlayOptionsBuilder = new GroundOverlayBuilder();
+    String groundOverlayId =
+        Convert.interpretGroundOverlayOptions(
+            platformGroundOverlay,
+            groundOverlayOptionsBuilder,
+            assetManager,
+            density,
+            bitmapDescriptorFactoryWrapper);
+    GroundOverlayOptions options = groundOverlayOptionsBuilder.build();
+    final GroundOverlay groundOverlay = googleMap.addGroundOverlay(options);
+    if (groundOverlay != null) {
+      GroundOverlayController groundOverlayController =
+          new GroundOverlayController(groundOverlay, platformGroundOverlay.getBounds() != null);
+      groundOverlayIdToController.put(groundOverlayId, groundOverlayController);
+      googleMapsGroundOverlayIdToDartGroundOverlayId.put(groundOverlay.getId(), groundOverlayId);
+    }
+  }
+
+  private void changeGroundOverlay(@NonNull Messages.PlatformGroundOverlay platformGroundOverlay) {
+    String groundOverlayId = platformGroundOverlay.getGroundOverlayId();
+    GroundOverlayController groundOverlayController =
+        groundOverlayIdToController.get(groundOverlayId);
+    if (groundOverlayController != null) {
+      Convert.interpretGroundOverlayOptions(
+          platformGroundOverlay,
+          groundOverlayController,
+          assetManager,
+          density,
+          bitmapDescriptorFactoryWrapper);
+    }
+  }
+
+  private void removeGroundOverlay(@NonNull String groundOverlayId) {
+    GroundOverlayController groundOverlayController =
+        groundOverlayIdToController.get(groundOverlayId);
+    if (groundOverlayController != null) {
+      groundOverlayController.remove();
```

Comment:
Does the order of these removals matter? if so please add a comment explaining the order.

Reply:
Order does not matter here